tests/integration/targets/vmware_guest_instant_clone/tasks/main.yml

```
@@ -0,0 +1,387 @@
+      password: "{{ vcenter_password }}"
+      name: "Instant_cloned_instance_uuid"
+      folder: "{{ f0 }}"
+      state: powered-off
+    register: poweroff_instant_clone_from_vm_instance_uuid
+
+  - debug:
+      var: poweroff_instant_clone_from_vm_instance_uuid
+
+  - name: Clean VM
+    vmware_guest:
+      validate_certs: false
+      hostname: "{{ vcenter_hostname }}"
+      username: "{{ vcenter_username }}"
+      password: "{{ vcenter_password }}"
+      name: "Instant_cloned_instance_uuid"
+      datacenter: "{{ dc1 }}"
+      state: absent
+    register: delete_instant_clone_from_vm_instance_uuid
+    ignore_errors: true
+
+  - debug: var=delete_instant_clone_from_vm_instance_uuid
+
+  - name: Instant Clone a VM with uuid
+    vmware_guest_instant_clone:
+      hostname: "{{ vcenter_hostname }}"
+      username: "{{ vcenter_username }}"
+      password: "{{ vcenter_password }}"
+      validate_certs: false
+      datacenter: "{{ dc1 }}"
+      folder: "{{ f0 }}"
+      datastore: "{{ rw_datastore }}"
+      host:  "{{ esxi1 }}"
+      name: "Instant_cloned_uuid"
+      uuid: "{{ uuid }}"
+      resource_pool: RP1
+    register: instant_clone_from_vm_uuid
+
+  - name: assert that changes were made
+    assert:
+      that:
+          - instant_clone_from_vm_uuid is changed
+
+  - name: set state to poweroff the VM
+    vmware_guest_powerstate:
+      validate_certs: false
+      hostname: "{{ vcenter_hostname }}"
+      username: "{{ vcenter_username }}"
+      password: "{{ vcenter_password }}"
+      name: Instant_cloned_uuid
+      folder: "{{ f0 }}"
+      state: powered-off
+    register: poweroff_instant_clone_from_vm_uuid
+
+  - debug:
+      var: poweroff_instant_clone_from_vm_uuid
+
+  - name: Clean VM
+    vmware_guest:
+      validate_certs: false
+      hostname: "{{ vcenter_hostname }}"
+      username: "{{ vcenter_username }}"
+      password: "{{ vcenter_password }}"
+      name: Instant_cloned_uuid
+      datacenter: "{{ dc1 }}"
+      state: absent
+    register: delete_instant_clone_from_vm_uuid
+    ignore_errors: true
+
+  - debug: var=delete_instant_clone_from_vm_uuid
+
+  - name: Instant clone in check mode
+    vmware_guest_instant_clone:
+      hostname: "{{ vcenter_hostname }}"
+      username: "{{ vcenter_username }}"
+      password: "{{ vcenter_password }}"
+      validate_certs: false
+      folder: "{{ f0 }}"
+      datastore: "{{ rw_datastore }}"
+      datacenter: "{{ dc1 }}"
+      host: "{{ esxi1 }}"
+      name: "cloned_vm_from_vm_check"
+      parent_vm: "test_vm1"
+      resource_pool: test_resource_001
+    check_mode: true
+    register: check_mode_instant_clone
+
+  - debug:
+      var: check_mode_instant_clone
+
+  - name:  idempotency check - VM name already exists
+    vmware_guest_instant_clone:
+      hostname: "{{ vcenter_hostname }}"
+      username: "{{ vcenter_username }}"
+      password: "{{ vcenter_password }}"
+      validate_certs: false
+      name: "cloned_vm_from_vm"
```

```suggestion
      name: "cloned_vm_from_vm"
```

In this test, a new instance clone is created because the cloned_vm_from_vm doesn't exist.